lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1449,7 +1449,10 @@ def install(collection, path, artifacts_manager):  # FIXME: mv to dataclasses?
     )
 
     if os.path.exists(b_collection_path):
-        shutil.rmtree(b_collection_path)
+        # See bug #79408 for details on ignoring_errors
+        shutil.rmtree(b_collection_path, ignore_errors=True)
+        if os.path.exists(b_collection_path):
+            raise AnsibleError("Failed to remove the existing collection at '%s'." % b_collection_path)
```

Comment:
```suggestion
        try:
            shutil.rmtree(b_collection_path)
        except Exception as e:
            raise AnsibleError(
                f"Failed to remove the existing collection at {to_text(b_collection_path)!r}. Ensure there are "
                f"not multiple instances of ansible-galaxy running in parallel. The error was: {e}"
            )
```

Reply:
@sivel it's probably a good idea to start recording exception causes by chaining them via `from e`. Also, maybe just catch an expected exception instead of suppressing everything?
```suggestion
        try:
            shutil.rmtree(b_collection_path)
        except OSError as os_err:
            raise AnsibleError(
                f"Failed to remove the existing collection at {to_text(b_collection_path)!r}. Ensure there are "
                f"not multiple instances of ansible-galaxy running in parallel. The error was: {e}"
            ) from os_err
```

Reply:
why we have `orig_exc` as part of all ansible exceptions, also it automatically adds that message so no need to add yourself.

Reply:
There's difference in presentation between explicit and implicit cause, though.